spec/services/record_declarations/started/early_career_teacher_spec.rb

```diff
@@ -2,51 +2,290 @@
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+      create(:participant_profile_state, :withdrawn, participant_profile: profile, cpd_lead_provider:)
     end
 
-    def user_profile
-      ect_profile
+    it "creates the declaration" do
+      expect { subject.call }.to change { ParticipantDeclaration.count }.by(1)
     end
   end
 
-  context "when user is for 2020 cohort" do
-    let!(:cohort_2020) { create(:cohort, start_year: 2020) }
-    let!(:school_cohort_2020) { create(:school_cohort, cohort: cohort_2020, school: ect_profile.school) }
+  context "when user profile ParticipantProfileState is withdrawn and also withdrawn on declaration date" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+      create(:participant_profile_state, :withdrawn, participant_profile: profile, cpd_lead_provider:, created_at: declaration_date_object - 2.days)
+    end
+
+    it "does not create the declaration" do
+      expect {
+        subject.call
+      }.to raise_error(ActionController::ParameterMissing, /Declaration must be before withdrawal date/)
+       .and not_change { ParticipantDeclaration.count }
+    end
+  end
+
+  context "when user profile ParticipantProfileState is deferred, but was active on declaration date" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+      create(:participant_profile_state, :deferred, participant_profile: profile, cpd_lead_provider:)
+    end
+
+    it "creates the declaration" do
+      expect { subject.call }.to change { ParticipantDeclaration.count }.by(1)
+    end
+  end
+
+  context "when incorrect course provided" do
+    let(:course_identifier) { "ecf-mentor" }
 
     before do
-      induction_programme.update!(school_cohort: school_cohort_2020)
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
     end
 
-    it "raises a ParameterMissing error" do
-      expect { described_class.call(params: ect_params) }.to raise_error(ActionController::ParameterMissing)
+    it "does not create the declaration" do
+      expect {
+        subject.call
+      }.to raise_error(ActionController::ParameterMissing, /The property '#\/course_identifier' must be an available course to '#\/participant_id'/)
+       .and not_change { ParticipantDeclaration.count }
+    end
+  end
+
+  context "when NPQ course provided" do
+    let(:course_identifier) { "npq-leading-teacher" }
+
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+    end
+
+    it "does not create the declaration" do
+      expect {
+        subject.call
+      }.to raise_error(ActionController::ParameterMissing, /The property '#\/course_identifier' must be an available course to '#\/participant_id'/)
+       .and not_change { ParticipantDeclaration.count }
+    end
+  end
+
+  context "when user is for 2020 cohort (ie NQT+1)" do
```

Comment:
Do we even need the anymore? i.e are we still receiving declarations for NQT+1 (could very well be)

Reply:
my understanding is we should never receive a declaration for an NQT+1. and if we do we should never create a declaration as no one should ever get paid for these